src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/OkHttpWebSocketWrapper.java

```diff
@@ -94,6 +94,10 @@ public void onClosing(WebSocket webSocket, int code, String reason) {
         @Override
         public void onFailure(WebSocket webSocket, Throwable t, Response response) {
             logger.error("WebSocket closed from an error: {}.", t.getMessage());
+            // Register a default error handler to avoid unhandled exceptions crashing the process
+            // This can happen if the connection closes from a network issue instead of the user calling stop()
+            // In which case no one is observing the closeSubject error.
+            closeSubject.doOnError((e) -> {});
             closeSubject.onError(new RuntimeException(t));
             onClose.invoke(null, t.getMessage());
             checkStartFailure();
```

Comment:
Is there a chance that checkStartFailure() ever gets called before the Completable is returned from start() meaning we'd need to do the same ting for startSubject?

Reply:
I think that is technically possible, but very unlikely in practice.

I'm not sure how the global error handler works in that case, because the startSubject will be observed by someone, they just haven't registered the handler by the time onError is called.

Reply:
If it's possible, we should handle it.

> I'm not sure how the global error handler works in that case, because the startSubject will be observed by someone, they just haven't registered the handler by the time onError is called.

We could test this by intentionally faulting the startSubject before returning. I expect it would also  be crashing. How would the runtime even know the startSubject will be observed by someone in the future?